### PR TITLE
Silo link RCD upgrade

### PR DESCRIPTION
**Pull request by Dennok**
## About The Pull Request
Silo link allow RCD take materials directly from ore silo.
RCD need to be linked to the silo as another new machinery.
Take only iron, 4 points per 1 sheet, like you use ore redemption machine, or fab and reload by hand.

## Why It's Good For The Game
Builders dream about it.

## Changelog
:cl:
add: Silo link RCD upgrade
add: RCD now check mats before construction wait.
fix: has_materials()  properly check materials.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
I don't know how I feel about this one personally

**Comment by AcapitalA** [comment]
On one hand, this is a pretty nice QoL feature which will save some of time.
On the other hand, it's power creep and most people will just simply drain the metal supply even faster.

Not sure on my opinion over this one either.

**Comment by Dennok** [comment]
> On one hand, this is a pretty nice QoL feature which will save some of time.
> On the other hand, it's power creep and most people will just simply drain the metal supply even faster.
> 
> Not sure on my opinion over this one either.

Yes, Now QM make a fortress near ore silo to prevent greedy engy to connect RCD to silo.

Anyway any who have acces to any already connected machinery can drain mats, just push Eject all button.

About power creep, it locked by 25000 points of research after all bluespace researches, and need some mats too.

I think it for long/super long rounds.

**Comment by FloranOtten** [comment]
Oh god. It's already hard enough to link a new lathe to the silo, now with the QM actively trying to prevent it?

**Comment by Dennok** [comment]
The sandbags, mining equipment, And need add Lazares injected goliath to vault.

**Comment by CPTANT** [comment]
This removes the bare minimum of restraints on just spamming everything you want with the RCD.

**Comment by Dennok** [comment]
> This removes the bare minimum of restraints on just spamming everything you want with the RCD.

I implement this in next PR's

And as i say before. you already can have near infinite material stach by crate with BoH fillen by mats. Or unwrenched linked ORM.

**Comment by Swagile** [comment]
You can already do this late game, it just makes it slightly easier.

Get a BoH, eject all metal / glass, put into BoH, you now have this except you need BoH tech to do it.

Considering this is gated behind research just like BoH and is more of a long term upgrade, its not that bad.

**Comment by FloranOtten** [comment]
> You can already do this late game, it just makes it slightly easier.
> 
> Get a BoH, eject all metal / glass, put into BoH, you now have this except you need BoH tech to do it.
> 
> Considering this is gated behind research just like BoH and is more of a long term upgrade, its not that bad.

The BoH at least takes up storage space. What's stopping the assistant from getting an RCD, destroying vault door then running around deconstructing every door?

**Comment by Dawson1917** [comment]
If someone wants to use an RCD for grief they can already take 2-3 stacks of metal/glass/reinforced glass from EVA, aux base, aux storage, or any lathe, and go wild. The only people I see getting use out of this would be engineers building space bases, extending the station, or working on the Build-Your-Own shuttle, which is good

**Comment by Swagile** [comment]
> 
> 
> > You can already do this late game, it just makes it slightly easier.
> > Get a BoH, eject all metal / glass, put into BoH, you now have this except you need BoH tech to do it.
> > Considering this is gated behind research just like BoH and is more of a long term upgrade, its not that bad.
> 
> The BoH at least takes up storage space. What's stopping the assistant from getting an RCD, destroying vault door then running around deconstructing every door?

The BoH can hold like 500 + materials, which can easily be used for grief already, yet it doesn't happen that much.

Same thing with this change; it's already a late game tech after all.

**Comment by CPTANT** [comment]
> 
> 
> > > You can already do this late game, it just makes it slightly easier.
> > > Get a BoH, eject all metal / glass, put into BoH, you now have this except you need BoH tech to do it.
> > > Considering this is gated behind research just like BoH and is more of a long term upgrade, its not that bad.
> > 
> > 
> > The BoH at least takes up storage space. What's stopping the assistant from getting an RCD, destroying vault door then running around deconstructing every door?
> 
> The BoH can hold like 500 + materials, which can easily be used for grief already, yet it doesn't happen that much.
> 
> Same thing with this change; it's already a late game tech after all.

This is still less ideal than a silo link because it requires all materials to be present at the time of gathering them. A silo link will always have everything.

**Comment by Dennok** [comment]
> > > > You can already do this late game, it just makes it slightly easier.
> > > > Get a BoH, eject all metal / glass, put into BoH, you now have this except you need BoH tech to do it.
> > > > Considering this is gated behind research just like BoH and is more of a long term upgrade, its not that bad.
> > > 
> > > 
> > > The BoH at least takes up storage space. What's stopping the assistant from getting an RCD, destroying vault door then running around deconstructing every door?
> > 
> > 
> > The BoH can hold like 500 + materials, which can easily be used for grief already, yet it doesn't happen that much.
> > Same thing with this change; it's already a late game tech after all.
> 
> This is still less ideal than a silo link because it requires all materials to be present at the time of gathering them. A silo link will always have everything.

As i say, you can yake unwrenched redemption mactine or build new machine/ fabricator and link it to take mats, only neet a some time(30 sec with rped) and clicks.

File: code/game/objects/items/RCD.dm [hunk 1]
```diff
@@ -190,6 +221,24 @@ RLD
 
 	to_chat(user, "<span class='notice'>You change \the [src]'s window mode to [window_type_name].</span>")
 
+/obj/item/construction/rcd/verb/toggle_silo_link_verb()
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/RCD.dm]: verbs out, buttons in

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]

File: code/game/objects/items/RCD.dm (hunk 1 above)
Comment (Dennok) [reply code/game/objects/items/RCD.dm]: Done

**Comment by ExcessiveUseOfCobblestone** [comment]
Yo is this on the radial? I forgot all about that :omegalul:

raddddd

**Comment by ExcessiveUseOfCobblestone** [comment]
> I don't know how I feel about this one personally
> 
> edit: the close was to announce it with its proper title

Same but it's a deep techweb investment so i'm okay with seeing how it plays out Live

**Comment by adamkad1** [comment]
People who want to grief dont need this. People who dont could use this. Saves a lot of time.  Its really not that big of a deal

**Pull request merged**